src/modules/quick_input_parser_module.f90

```diff
@@ -14,70 +14,67 @@ module quick_input_parser_module
 
     contains
 
-        subroutine read_float_keyword(keywdline, keyword, val)
+        subroutine checkformat(i,line,keyword)
             implicit none
-            character(len=*), intent(in) :: keywdline
+            integer, intent(inout) :: i
+            character(len=*), intent(in) :: line
+            character(len=*), intent(in) :: keyword
+
+            i = index(line, trim(keyword)//trim('='))
```

Comment:
You don't need the second `trim` here because `'='` is guaranteed not to have any trailing blanks.